lib/gis/lrand48.c

```diff
@@ -41,22 +45,36 @@ static const uint32 b0 = 0xB;
 
 static int seeded;
 
+#ifdef HAVE_PTHREAD
+static pthread_mutex_t lrand48_mutex = PTHREAD_MUTEX_INITIALIZER;
+#endif
+
 #define LO(x) ((x) & 0xFFFFU)
 #define HI(x) ((x) >> 16)
 
 /*!
  * \brief Seed the pseudo-random number generator
  *
  * \param[in] seedval 32-bit integer used to seed the PRNG
+ *
+ * This function is thread-safe.
```

Comment:
Something like the following should indicate that this applies only for certain builds. I don't know if you can actually have a combination when the thread safety matters while not having pthreds.

```suggestion
 * This function is thread-safe (if threading is enabled, i.e., `HAVE_PTHREAD` is defined).
```

or: ...if pthreads is available,...

Reply:
Disagree. This makes no sense. Of course there is no thread safety when code is compiled without threads! This is a programmer documentation that is meant to signal other developers that it is safe to call this function from threaded code. Attempts to compile core GRASS without `pthreads` and calling from a module with `pthreads` should not be considered in documentation of each function.

Reply:
Fair enough.